### ISSUE TITLE
500 internal errors infrequently

### ISSUE THREAD

**Issue by schveiguy** [vibe-d/vibe.d#2555]
I don't know where to start on this, but I'll start here, because it seems like vibe is either muffling the true error, or it is truly an error in vibe.

I am getting infrequent, but noticeable, errors that go away with a page refresh. In my code, I am printing out any error stack traces, and when an exception is thrown in MY code, I see the full thing. What I get for these (with logging set to trace) is:

```
route match: /management/sales/translate_option -> GET * []
returning Connection connection 2 of 6
returning RedisConnection connection 0 of 3
Socket 12, read 7 bytes: ok
Socket 13, read 1685 bytes: ok
Error occurred while accessing /management/sales/translate_option: code 500, (Internal Server Error)
Stack Trace: Internal Server Error
```

Is there anything anyone can deduce from this? Do I need to enable more debugging flags?

**Comment by schveiguy** [comment]
FYI, I'm using Redis for session storage, which is why you see that RedisConnection.

**Comment by schveiguy** [comment]
This is the start of my handleError function that prints that:

```d
void handleError(HTTPServerRequest req, HTTPServerResponse res, HTTPServerErrorInfo err)
{
    logDiagnostic("Error occurred while accessing %s: code %s, (%s)\nStack Trace: %s\n%s", req.path, err.code, httpStatusText(err.code), err.message, err.debugMessage);
```

**Comment by s-ludwig** [comment]
There is `HTTPServerOption.errorStackTraces` that needs to be set in `HTTPServerSettings` for the stack traces to be actually stored (to avoid exposing memory addresses by default).

**Comment by schveiguy** [comment]
Thanks, I'll try it again. I get this regularly enough that it should happen quickly.

**Comment by schveiguy** [comment]
ugh, I did this (somewhat) to myself -- I switched to using `dub -b plain`, which disables `-debug`. So my perception was that I wasn't doing anything different, but really I was disabling the stack traces that I would normally see.

The real stack trace is in mysql-native, and it may be something that has nothing to do with vibe-core, and it may not. The message says "Server packet out of order". I may have to get more familiar with it to figure out why this is happening, but I will have some feedback later. I definitely need this fixed, so I will hunt it down.
